docs/content/querying/lookups.md

```diff
@@ -296,3 +296,252 @@ To test this setup, you can send key/value pairs to a kafka stream via the follo
+
+```json
+{
+    "type": "confidential_jdbc",
+    "auth": "/etc/jdbc.customer1",
+    "table": "sites_updated",
+    "key": "site_id",
+    "value": "site_name"
+}
+```
+
+This will replace the `site_id_customer1` lookup in the `realtime_customer1` with the definition above.
+
+## Get Lookup
+A `GET` to a particular lookup extractor factory is accomplished via `/druid/coordinator/v1/lookups/{tier}/{id}`
+
+Using the prior example, a `GET` to `/druid/coordinator/v1/lookups/realtime_customer2/site_id_customer2` should return
+
+```json
+{
+    "type": "confidential_jdbc",
+    "auth": "/etc/jdbc.customer2",
+    "table": "sites",
+    "key": "site_id",
+    "value": "site_name"
+}
+```
+
+## Delete Lookup
+A `DELETE` to `/druid/coordinator/v1/lookups/{tier}/{id}` will remove that lookup from the cluster.
+
+## List tier names
+A `GET` to `/druid/coordinator/v1/lookups` will return a list of known tier names in the dynamic configuration.
+To discover a list of tiers currently active in the cluster **instead of** ones known in the dynamic configuration, the parameter `discover=true` can be added as per `/druid/coordinator/v1/lookups?discover=true`.
+
+## List lookup names
+A `GET` to `/druid/coordinator/v1/lookups/{tier}` will return a list of known lookup names for that tier.
+
+# Internal API
+
+The Peon, Router, Broker, and Historical nodes all have the ability to consume lookup configuration.
+There is an internal API these nodes use to list/load/drop their lookups starting at `/druid/listen/v1/lookups`.
+These follow the same convention for return values as the cluster wide dynamic configuration.
+Usage of these endpoints is quite advanced and not recommended for most users.
+The endpoints are as follows:
+
+## Get Lookups
+
+A `GET` to the node at `/druid/listen/v1/lookups` will return a json map of all the lookups currently active on the node.
+The return value will be a json map of the lookups to their extractor factories.
+
+```json
+
+{
+    "some_lookup_name": {
+        "type": "simple_json",
+        "uri": "http://some.host.com/codes.json"
+    }
+}
+
+```
+
+## Get Lookup
+
+A `GET` to the node at `/druid/listen/v1/lookups/some_lookup_name` will return the LookupExtractorFactory for the lookup identified by `some_lookup_name`.
+The return value will be the json representation of the factory.
+
+```json
+{
+    "type": "simple_json",
+    "uri": "http://some.host.com/codes.json"
+}
+```
+
+## Bulk Add or Update Lookups
+
+A `POST` to the node at `/druid/listen/v1/lookups` of a JSON map of lookup names to LookupExtractorFactory will cause the service to add or update its lookups.
+The return value will be a JSON map in the following format:
+
+```json
+{
+    "status": "accepted",
+    "failedUpdates", {}
+}
+
+```
+
+If a lookup cannot be started, or is left in an undefined state, the lookup in error will be returned in the `failedUpdates` field as per:
```

Comment:
This is a function of the API when pinging a historical directly; it is part of the internal API